src/main/java/com/deviceinsight/kafka/health/KafkaConsumingHealthIndicator.java

```diff
@@ -60,15 +66,18 @@
 
 	private KafkaCommunicationResult kafkaCommunicationResult;
 
+	@Autowired
+	private MeterRegistry meterRegistry;
+
 	public KafkaConsumingHealthIndicator(KafkaHealthProperties kafkaHealthProperties,
-			Map<String, Object> kafkaConsumerProperties, Map<String, Object> kafkaProducerProperties) {
+			Map<String, Object> kafkaConsumerProperties, Map<String, Object> kafkaProducerProperties,
+			@Autowired MeterRegistry meterRegistry) {
```

Comment:
The `@Autowired` annotation is also unnecessary here. This class should be initialized explicitly.

Reply:
what i'm trying to do here is create a bean in `MetricsConfig` and autowiring it here through construtor injection. Let me know if you want me to change this approach.

Reply:
Yes, I understand. But the class should be instantiated as described in the README rather than by autowiring dependencies, since this might cause issues for some projects that have multiple `MeterRegistry` beans, for example.